static/js/components/SaveCandidateButton.jsx

```diff
@@ -0,0 +1,210 @@
+import React, { useState, useEffect, useRef } from "react";
+import PropTypes from "prop-types";
+import { useDispatch } from "react-redux";
+import Dialog from "@material-ui/core/Dialog";
+import DialogContent from "@material-ui/core/DialogContent";
+import DialogTitle from "@material-ui/core/DialogTitle";
+import Checkbox from "@material-ui/core/Checkbox";
+import Button from "@material-ui/core/Button";
+import ButtonGroup from '@material-ui/core/ButtonGroup';
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+import ArrowDropDownIcon from "@material-ui/icons/ArrowDropDown";
+import ClickAwayListener from "@material-ui/core/ClickAwayListener";
+import Grow from "@material-ui/core/Grow";
+import Paper from "@material-ui/core/Paper";
+import Popper from "@material-ui/core/Popper";
+import MenuItem from "@material-ui/core/MenuItem";
+import MenuList from "@material-ui/core/MenuList";
+import { useForm, Controller } from "react-hook-form";
+
+import * as sourceActions from "../ducks/source";
+import FormValidationError from "./FormValidationError";
+
+
+const options = ["Save as Source", "Save (select groups)"];
+
+const SaveCandidateButton = ({ candidate, userGroups }) => {
+  // Dialog logic:
+
+  const dispatch = useDispatch();
+  const [dialogOpen, setDialogOpen] = useState(false);
+
+  const { handleSubmit, errors, reset, control, getValues } = useForm();
+
+  useEffect(() => {
+    reset({
+      group_ids: userGroups.map((userGroup) => candidate.passing_group_ids.includes(userGroup.id))
+    });
+  }, [reset, userGroups, candidate]);
+
+  const handleClickOpenDialog = () => {
+    setDialogOpen(true);
+  };
+
+  const handleCloseDialog = () => {
+    setDialogOpen(false);
+  };
+
+  const validateGroups = () => {
+    const formState = getValues({ nest: true });
+    return formState.group_ids.filter((value) => Boolean(value)).length >= 1;
+  };
+
+  const onSubmit = async (data) => {
+    data.id = candidate.id;
+    const groupIDs = userGroups.map((g) => g.id);
+    const selectedGroupIDs = groupIDs.filter((ID, idx) => data.group_ids[idx]);
+    data.group_ids = selectedGroupIDs;
+    const result = await dispatch(sourceActions.saveSource(data));
```

Comment:
Should we disable save button until action has been completed?

Should a message be displayed next to the button on error, or OK to rely on notification?

Reply:
As for your second comment, let's rely on the notification -- that's exactly what they're for! :)

Reply:
Done

Reply:
Yes, my concern was that the user would be focusing on the button, and looking at the form for feedback.  But, I agree that the notifications are prominent enough not to miss.